File: leetcode/Python/sum-of-root-to-leaf-binary-numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumRootToLeaf(self, root: Optional[TreeNode]) -> int:
        sol, dq , cur, val = 0, collections.deque(), None, 0
        dq.append([root, 0])
        while dq:
            cur = dq.popleft()
            val = (cur[1]<<1) + cur[0].val
            if cur[0].left is None and cur[0].right is None:
                sol += val
            if cur[0].left:
                dq.append([cur[0].left, val])
            if cur[0].right:
                dq.append([cur[0].right, val])
        return sol

File: leetcode/C++/sum-of-root-to-leaf-binary-numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sumRootToLeaf(TreeNode* root) {
        TreeNode* node=nullptr;
        int val = 0, sol = 0;
        queue<pair<TreeNode*,int>> q;
        q.push({root, 0});
        while(!q.empty()){
            node = q.front().first; val = ((q.front().second)<<1) + node->val; q.pop();
            if(node->left == nullptr && node->right == nullptr) sol += val;
            if(node->left) q.push({node->left, val});
            if(node->right) q.push({node->right, val});
        }
        return sol;
    }
};